### PR TITLE
DAOS-7821 cq: Improve pylint coverage and fix issues.

### PR DESCRIPTION
**Pull request by ashleypittman**
- DAOS-7821 cq: Improve pylint checking and resolve warnings.
- Resolve some spellings.

### CODE REVIEW

File: src/tests/ftest/util/data_mover_test_base.py [hunk 1]
```diff
@@ -594,16 +592,15 @@ def dataset_verify(self, obj_list, cont, num_objs, num_dkeys,
...
                     for data_idx in range(num_extents):
                         data_val = str(data_idx % 10)
                         data = data_size * data_val
                         actual_idx = str(actual_data[data_idx].decode())
                         if data != actual_idx:
                             self.log.info("Expected:\n%s\nBut got:\n%s",
-                                data[:100] + "...",
-                                actual_idx + "...")
+                                          data[:100] + "...",
+                                          actual_idx + "...")
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/tests/ftest/util/data_mover_test_base.py:603]: I'm not writing the style guide, python is reasonably well defined here, I'm just making sure we're using the tools available to us.

Personally I prefer after the change.

File: src/tests/ftest/util/rebuild_test_base.py [hunk 2]
```diff
@@ -185,11 +186,10 @@ def execute_rebuild_test(self, create_container=True):
...
         # clear container status for the RF issue
-        self.daos_cmd.container_set_prop(
-                      pool=self.pool.uuid,
-                      cont=self.container.uuid,
-                      prop="status",
-                      value="healthy")
+        self.daos_cmd.container_set_prop(pool=self.pool.uuid,
+                                         cont=self.container.uuid,
+                                         prop="status",
+                                         value="healthy")
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/tests/ftest/util/rebuild_test_base.py:192]: Here with the ( on the end of the line I would say the old version was obviously wrong.

File: utils/cq/daos_pylint.py [hunk 3]
```diff
@@ -190,6 +187,16 @@ def is_scons_file(filename):
             # There may be more files needed here, but just this one is reporting errors.
             if filename.endswith('site_scons/site_tools/protoc/__init__.py'):
                 return True
+            if filename.endswith('site_scons/stack_analyzer.py'):
+                return True
+            # Needs more work yet, partly on spellings.  Another issue is that in GitHub actions
+            # pylint is called on all files in the tree concurretnly so it can resolve calls to
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply utils/cq/daos_pylint.py:193]: Will fix.  Ironic that there is a spelling mistake here.

File: utils/cq/daos_pylint.py [hunk 4]
```diff
@@ -387,6 +392,22 @@ def main():
...
+    if spellings and args.rcfile == rcfile:
+        # pylint: disable-next=consider-using-with
+        rc_tmp = tempfile.NamedTemporaryFile(mode='w+', prefix='pylintrc')
+        with open(rcfile) as src_file:
+            rc_tmp.write(src_file.read())
+        rc_tmp.flush()
+        rc_tmp.write('[SPELLING]\n')
+        rc_tmp.write('spelling-dict=en_US\n')
+        rc_tmp.write('spelling-private-dict-file=utils/cq/words.dict\n')
+        rc_tmp.flush()
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply utils/cq/daos_pylint.py:408]: We might.  I see what you're saying but I'm not inclined to put too much effort into it.

**Review by daltonbohning — DISMISSED** [review]

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/util/data_mover_test_base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/tests/ftest/util/data_mover_test_base.py]: After speaking with Dalton and consulting pep-8 there's another style here which I wasn't aware of and is more appropriate for this case so I've fixed this code a different way.

File: src/tests/ftest/util/rebuild_test_base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/tests/ftest/util/rebuild_test_base.py]: I was plain incorrect here, this style is allowed, the warning was not about the style but the level of indentation which I have now fixed.

[the remark this answers is not in the document]
Comment (daltonbohning) [reply src/tests/ftest/util/rebuild_test_base.py]: Thanks :)

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**